documentation/modules/cruise-control/con-rebalance-goals-proposals.adoc

```diff
@@ -0,0 +1,180 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-rebalance-goals-proposals-{context}']
+= Rebalance goals and proposals
+
+To rebalance a Kafka cluster, Cruise Control uses _rebalance goals_ to generate _rebalance proposals_. 
+
+[discrete]
+== Rebalance goals
+
+Rebalance goals specify targets for resource distribution and utilization across a Kafka cluster. 
+They are organized in a customizable hierarchy, from high priority goals to low priority goals. 
+You can designate certain rebalance goals as _hard goals_ that must be satisfied--for more information, see link:#soft-hard-goals[Soft goals and hard goals].
+
+With a few exceptions, cluster rebalancing in {ProductName} supports the same goals as the Kafka Cruise Control project (where the term "optimization goals" is used). 
+Following are the supported goals, which you can reorder in priority, enable, or disable:
+
+. Rack-awareness
+. Replica capacity
+. Capacity
+.. Disk capacity
+.. Network inbound capacity
+.. Network outbound capacity
+.. CPU capacity
+. Replica distribution
+. Potential network output
+. Resource distribution
+.. Disk utilization distribution
+.. Network inbound utilization distribution
+.. Network outbound utilization distribution
+.. CPU utilization distribution
+. Leader bytes-in rate distribution
+. Topic replica distribution
+. Leader replica distribution
+. Preferred leader election
+. Intra-broker disk capacity
+. Intra-broker disk usage distribution    
```

Comment:
There is still some outstanding work needed to support the CPU capacity,  CPU utilization distribution,  Intra-broker disk capacity, and Intra-broker disk usage distribution goals, so those can be removed from this list

Reply:
OK. I've commented out those goals from the list.

Reply:
At the start, we state that CC is used for: 

"_Monitoring the CPU_, disk, and network load of Kafka brokers."

Can that stay?

Reply:
I think we can leave it as it is as a summary for CC. CC will still monitor CPU when running on k8s, it's just the values will not be accurate in all situations (at least until we get the metric reporter patch in)